Save the inspirational images

Simeon10 opened the pull request:
I wrote all the code for the migration as well, just not 100% sure on where exactly it goes. Right now it's in the inspirational images handler witht he "migrate_inspirational_images" route. If somebody could let me know then I could fix that and merge if all else is fine. The code also assumes that "solved circular dependency" has been merged. So this can only merge once the other one merges.

Simeon10 committed:
First submit

I wrote all the code for the migration as well, just not 100% sure on where exactly it goes. Right now it's in the inspirational images handler witht he "migrate_inspirational_images" route. If somebody could let me know then I could fix that and merge if all else is fine. The code also assumes that "solved circular dependency" has been merged. So this can only merge once the other one merges.

bonnieli commented on client/src/components/CreateContestForm.js:
```diff
@@ -71,7 +71,7 @@ export default function CreateContestForm() {
     const [time, setTime] = useState(new Date().toTimeString().substr(0,5))
     const [timeZone, setTimeZone] = useState('PDT')
-    const [images, setImages] = useState('')
+    const [inspirationalImages, setImages] = useState(new Array())
```
```suggestion
    const [inspirationalImages, setImages] = useState([])
```

you can simple just initiate an empty array :)

bonnieli commented on client/src/components/CreateContestForm.js:
```diff
@@ -104,12 +104,34 @@ export default function CreateContestForm() {
-    'tattoo-11.png', 'tattoo-12.png', 'tattoo-13.png', 'tattoo-14.png']
-
+    useEffect(() => { // Only runs once when first rendering
+        getInspirationalImages((data) => {
```
```suggestion
        getInspirationalImages(createGridListTiles, (error) => {
        ...
```

just a more simple and shorter way of writing this out

bonnieli commented on client/src/components/CreateContestForm.js:
```diff
@@ -104,12 +104,34 @@ export default function CreateContestForm() {
+            const imageURL = imageMap.get('image_link')
+            const imageKey = imageMap.get('id')
+            images.push(
+                <GridListTile key={imageKey} cols={1}>
```
you can probably simplify this code by just mapping through `data`

```
const = images [
  data.map(image => (
    <GridListTile key={image.id} cols={1}>
        <img src={image.image_link} alt={image.image_link} />
    </GridListTile>
  ))
]
```

Simeon10 replied:
I just get "data.map" is not a function

bonnieli replied:
oh that's weird, is data not an array?

Reference on how React usually uses the map function https://github.com/hatchways/team-ninja-turtles/blob/dev/client/src/pages/DiscoveryPage.js#L100

regardless, this isn't too big of an issue. it would just make the code more clean

bonnieli commented on server/api/inspirational_images_handler.py:
```diff
@@ -0,0 +1,46 @@
+def migrateImages():
+    s3_resource = boto3.resource('s3')
+    my_bucket = s3_resource.Bucket(S3_BUCKET)
+    summaries = my_bucket.objects.all()
```
this is good, instead of grabbing all the objects in here, if it's possible, limit it to a folder just incase all their objects in that bucket is like 100 diff images 

if you limit so that developer has a folder called `/inspo_images` for example, you only grab the images from inside that folder

bonnieli reviewed (commented):
overall good, just address some comments there.

in terms of the area for migrations, i think it's fine to leave it in there but if you want to do it best practice, we can take advantage of our migration architecture. https://github.com/hatchways/team-ninja-turtles/tree/dev/server/migrations/versions In this folder, these files are automatically created, but if you were to create a custom migration, you would follow this similar format and port over your code there under the `upgrade` function. this way, it will only ever get run once because that's all that's needed

Simeon10 committed:
Merge branch 'dev' into Save-the-inspirational-images

Simeon10 committed:
Updated based on suggestions

Simeon10 committed:
Select Images; Get Images for contest; Updated Model

I will create another commit shortly with the migration. So far this code itself works on my end :)

Simeon10 committed:
Update README.md

Simeon10 committed:
Added button to add images

You can now paste in a link to add external inspirational images to your contest.

Simeon10 committed:
Merge pull request #79 from hatchways/Select-Inspirational-Images

Select Images; Get Images for contest; Updated Model

edyhtan committed:
Added migration files.

edyhtan commented:
Added migration files.

Simeon10 committed:
Merge branch 'dev' into Save-the-inspirational-images
